src/CallMessage.tsx

```diff
@@ -0,0 +1,314 @@
+    };
+
+    NativeEventEmitter.addListener(
+      Constants.ScopeCallMessage,
+      this._handleNativeEvent
+    );
+  }
+
+  private _handleNativeEvent = (
+    nativeCallMessageEvent: NativeCallMessageEvent
+  ) => {
+    const { type } = nativeCallMessageEvent;
+
+    const handler = this._nativeEventHandler[type];
+    if (typeof handler === 'undefined') {
+      throw new Error(
+        `Unknown call message event type received from the native layer: "${type}"`
+      );
+    }
+
+    handler(nativeCallMessageEvent);
+  };
+
+  private _handleFailureEvent = (
+    nativeCallMessageEvent: NativeCallMessageEvent
+  ) => {
+    if (nativeCallMessageEvent.type !== Constants.CallEventMessageFailure) {
+      throw new Error(
+        'Incorrect "callMessage#Failure" handler called for type' +
+          `"${nativeCallMessageEvent.type}`
+      );
+    }
+
+    const { message, code } = nativeCallMessageEvent.error;
+    const error = constructTwilioError(message, code);
+    this.emit(CallMessage.Event.Failure, error);
+  };
+
+  private _handleReceivedEvent = (
+    nativeCallMessageEvent: NativeCallMessageEvent
+  ) => {
+    if (nativeCallMessageEvent.type !== Constants.CallEventMessageReceived) {
+      throw new Error(
+        'Incorrect "callMessage#Received" handler called for type' +
+          `"${nativeCallMessageEvent.type}`
+      );
+    }
+
+    const { callMessageSID, callMessage: callMessageInfo } =
+      nativeCallMessageEvent;
+
+    const callMessage = new CallMessage(callMessageInfo);
+
+    this.emit(CallMessage.Event.Received, callMessageSID, callMessage);
+  };
+
+  private _handleSentEvent = (
+    nativeCallMessageEvent: NativeCallMessageEvent
+  ) => {
+    if (nativeCallMessageEvent.type !== Constants.CallEventMessageSent) {
+      throw new Error(
+        'Incorrect "callMessage#Sent" handler called for type' +
+          `"${nativeCallMessageEvent.type}"`
+      );
+    }
+
+    const { callMessageSID } = nativeCallMessageEvent;
+
+    this.emit(CallMessage.Event.Sent, callMessageSID);
+  };
+
+  getContent(): string | undefined {
+    return this._content;
+  }
+
+  getContentType(): string | undefined {
+    return this._contentType;
+  }
+
+  getMessageType(): string | undefined {
+    return this._messageType;
+  }
+
+  getMessageSID(): string | undefined {
+    return this._messageSID;
+  }
+}
+
+/**
+ * Namespace for enumerations and types used by
+ * {@link (CallMessage:class) | CallMessage objects}.
+ *
+ * @remarks
+ *  - See also the {@link (CallMessage:class) | CallMessage class}.
+ *  - See also the {@link (CallMessage:interface) | CallMessage interface}.
+ *
+ * @public
+ */
+export namespace CallMessage {
+  /**
+   * Enumeration of all event strings emitted by {@link (CallMessage:class)} objects.
+   */
+  export enum Event {
+    /**
+     * Raised when call message fails.
+     * See {@link (CallMessage:interface).(addListener:2)}.
```

Comment:
The `addListener` numbers should be `1`, `2`, and `3`.

Reply:
I noticed you started at 2 for `src/Call.tsx`, is this a typo? 
https://github.com/twilio/twilio-voice-react-native/blob/main/src/Call.tsx#L848

Reply:
Yes, that's an error. We will have to fix that, I will file a ticket.